eng/scripts/SetTestPipelineVersion.ps1

```diff
@@ -0,0 +1,35 @@
+# Overides the project file and CHANGELOG.md for the template project using the next publishable version
+# This is to help with testing the release pipeline.
+
+. "${PSScriptRoot}\..\common\scripts\common.ps1"
+$latestTags = git tag -l "@azure/template*beta*"
+$semVars = @()
+
+$packageDirectory = "${PSScriptRoot}\..\..\sdk\template\template"
+$templatePackageFile = "${packageDirectory}\package.json"
+$changeLogFile = "${packageDirectory}\CHANGELOG.md"
+
+
+Foreach ($tags in $latestTags)
+{
+  $semVars += $tags.Replace("@azure/template_", "")
+}
+
+$semVarsSorted = [AzureEngSemanticVersion]::SortVersionStrings($semVars)
+LogDebug "Last Published Version $($semVarsSorted[0])"
+
+$newVersion = [AzureEngSemanticVersion]::ParseVersionString($semVarsSorted[0])
+$newVersion.IncrementAndSetToPrerelease()
+LogDebug "Version to publish [ $($newVersion.ToString()) ]"
+
+$packageFileContent = Get-Content -Path $templatePackageFile | ConvertFrom-Json
```

Comment:
Why aren't we using the versioning and change log scripts to manipulate these values?

Reply:
Those scripts don't provide a way to overwrite the version. What we currently have is a way to increment the version based on the current version in source which is not what we want. Also the UpdateChangelog Logic does not have a way to add content to the release note entry. I guess I could move the logic into the Update-Change-Log.ps1 script, but that will require further refactoring and testing which will add more blockage to the things am working on considering that this work is supposed to help with testing and unblocking of other work.

Reply:
For what it is worth I think we need a way to do this as this is a scenario that devs often need to do before they do a release. They will often time set the version and need to stamp in the release date in the change log. This is why I filed https://github.com/Azure/azure-sdk-for-net/issues/11546 but I've never gotten around to adding that support. If it isn't too difficult I'd prefer we add that support in the set version and change log scripts instead of implementing it independently as it will just be another set of logic we have to update if we ever change what things we want to update in those other scripts.

Reply:
Can we take care of that later. A lot of things are currently blocked and waiting on other stuff. Implementation for updating package version is in js, python and PowerShell for the various language. Trying to get it all unified now will just lead to more blocked work for longer time. It will be more realistic to get the eng common workflow working so that many of the blocked work can be completed.

Reply:
Yes we can do that in a follow-up.

Reply:
Can I just update Azure/azure-sdk-for-net#11546 to capture all the work that need to be don unifying the update scripts rather than creating a new issue.

Reply:
Sure you can update that issues but I think we need an issue tracking this for all the languages.